### PR TITLE
Use (Git) bash scripts for wrapping clang

### PR DESCRIPTION
**Pull request by anakrish**
This simplifies the Windows build experiences.
The entire windows folder can now be removed.

### CODE REVIEW

File: cmake/maybe_build_using_clangw.cmake [hunk 1]
```diff
@@ -48,31 +44,30 @@ function(maybe_build_using_clangw OE_TARGET)
 
     # Setup library tool variables
     set(CMAKE_C_CREATE_STATIC_LIBRARY 
-        "\"${CMAKE_BINARY_DIR}/windows/clangw/llvm-arw.exe\" qc <TARGET> <OBJECTS>" 
+        "bash \"${PROJECT_SOURCE_DIR}/scripts/llvm-arw\" \"qc <TARGET> <OBJECTS>\""
```
Comment (andschwa) [inline cmake/maybe_build_using_clangw.cmake]: We should probably be using `${BASH}` (already set in another file from `find_program` call, and we might need to move that upwards).

File: scripts/clangw [hunk 2]
```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
```
Comment (andschwa) [inline scripts/clangw]: Scary! I might personally prefer the C++ project, but that's because Bash is really scary code to write and validate.

**Comment by anakrish** [comment]
bors try

File: scripts/clangw (hunk 2 above)
Comment (andschwa) [reply scripts/clangw]: That's fair enough.

**Comment by anakrish** [comment]
@andschwa 
Testing this out in azure windows vm now.
Will close it out later today.

**Comment by anakrish** [comment]
Added block label to do more qualification

**Comment by andschwa** [comment]
@anakrish I am just testing bors, don't mind me:

bors r+

**Comment by andschwa** [comment]
Oh! I know why that didn't work, it's not rebased on master. Nevermind, I'll spin up another test.

**Comment by anakrish** [comment]
@andschwa I like the new "do not merge" tag.
I will let the CI settle down a bit with the Windows support before pushing this in.

**Comment by anakrish** [comment]
bors try

**Comment by anakrish** [comment]
@johnkord @andschwa Do you know why the following error happens only on one platform?
In scripts/clangw line 70:

        clang -target x86_64-pc-linux ${args[@]}

                                      ^-- SC2068: Double quote array expansions, otherwise they're like $* and break on spaces.

https://oe-jenkins.eastus.cloudapp.azure.com/blue/organizations/jenkins/Bors/detail/trying/747/pipeline

**Comment by anakrish** [comment]
bors try

**Comment by anakrish** [comment]
bors try

**Comment by andschwa** [comment]
> @johnkord @andschwa Do you know why the following error happens only on one platform?

Only answer I got for you is different ShellCheck versions.

**Comment by anakrish** [comment]
bors try

**Comment by anakrish** [comment]
@andschwa @johnkord @ionutbalutoiu @aserdean 

Is there some way to properly fix the shell-check warnings?
After playing with it for couple of hours, I cannot come up with the correct way to do it so that shell-check is happy.

**Comment by andschwa** [comment]
You can add an ignore like [here](https://github.com/Microsoft/openenclave/blob/a6dc781cb0eb1597901e5ef5e1b950d3c0423e3f/scripts/format-code#L130).

**Comment by ionutbalutoiu** [comment]
> > @johnkord @andschwa Do you know why the following error happens only on one platform?
> 
> Only answer I got for you is different ShellCheck versions.

Yep, I took a look into this and that's the case. See my https://github.com/Microsoft/openenclave/issues/1526#issuecomment-469733696, and let me know if my suggestion sounds good to you in order to get the inconsistency solved.

**Comment by johnkord** [comment]
@anakrish @ionutbalutoiu What's the status of this PR? Do we want to merge this? Should this be scheduled for 0.6?

**Comment by anakrish** [comment]
@johnkord This is Windows build experience. Applicable for v0.6

**Comment by ionutbalutoiu** [comment]
@anakrish The CI should report consistent linter errors since https://github.com/Microsoft/openenclave/issues/1526 was fixed.

When you get the chance, you can rebase on top of latest master and have this ready to be merged.

**Comment by andschwa** [comment]
From triage meeting: @anakrish Can you update us on this? Thanks!

**Comment by anakrish** [comment]
@andschwa I will wait for Bruce/Mike's work to go in before working on this again.
Only @anitagov needs this and she can pull in the necessary changes from this branch onto hers if needed.

**Comment by anitagov** [comment]
I cherry-picked these changes into my branch anitagov.libc.win and my issue with not being able to get all the tests to compile on Windows was indeed resolved. So, will take care of checking this into the master whenever possible.

**Comment by andschwa** [comment]
Cool, I'll close this then.

**Comment by anitagov** [comment]
We should probably keep this open until these changes get into master. Please do not delete the branch in any case.

This change does not work in Release mode but works in Debug mode. Need to debug this further.

**Comment by anitagov** [comment]
bors try

**Pull request closed without merging**